core/trino-spi/src/main/java/io/trino/spi/type/TypeOperators.java

```diff
@@ -192,14 +192,14 @@ private OperatorAdaptor getOperatorAdaptor(Type type, Optional<SortOrder> sortOr
     private class OperatorAdaptor
     {
         private final OperatorConvention operatorConvention;
-        private MethodHandle adapted;
+        private volatile MethodHandle adapted;
 
         public OperatorAdaptor(OperatorConvention operatorConvention)
         {
             this.operatorConvention = operatorConvention;
         }
 
-        public synchronized MethodHandle get()
+        public MethodHandle get()
         {
             if (adapted == null) {
                 adapted = adaptOperator(operatorConvention);
```

Comment:
should the assignment itself be in critical section or it doesn't matter if `adaptOperator` is called couple of times by concurrent threads?

Reply:
It doesn't seem to be all stateless operations here, would be safer to use double lock checking